.github/workflows/record.yml

```diff
@@ -0,0 +1,286 @@
+            uses: actions/setup-go@v4
+            with:
+                go-version: ${{ env.GOVERSION }}
+          - name: Install pulumictl
+            uses: jaxxstorm/action-install-gh-release@v1.1.0
+            with:
+                repo: pulumi/pulumictl
+          - name: Install Pulumi CLI
+            uses: pulumi/action-install-pulumi-cli@v1.0.1
+          - name: Setup Python
+            uses: actions/setup-python@v2
+            with:
+                python-version: ${{ env.PYTHONVERSION }}
+          - name: Setup DotNet
+            uses: actions/setup-dotnet@v1
+            with:
+                dotnet-version: ${{ env.DOTNETVERSION }}
+          - name: Setup Node
+            uses: actions/setup-node@v1
+            with:
+                node-version: ${{ env.NODEVERSION }}
+                registry-url: https://registry.npmjs.org
+          - name: Setup Java
+            if: matrix.language == 'java'
+            uses: actions/setup-java@v3
+            with:
+                cache: gradle
+                distribution: temurin
+                java-version: ${{ env.JAVAVERSION }}
+          - name: Setup Gradle
+            if: matrix.language == 'java'
+            uses: gradle/gradle-build-action@v2
+            with:
+                gradle-version: "7.6"
+          - name: Install Yarn
+            run: curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.13.0
+          - name: Update PATH for Yarn
+            run: |
+                echo "$HOME/.yarn/bin" >> $GITHUB_PATH
+                echo "$HOME/.config/yarn/global/node_modules/.bin" >> $GITHUB_PATH
+          - name: Build SDK
+            run: make build_${{ matrix.language }}
+          - name: Check worktree clean
+            run: ./ci-scripts/ci/check-worktree-is-clean
+          - name: Compress NodeJS SDK folder
+            if: matrix.language == 'nodejs'
+            run: tar -zcf ${{ matrix.language }}.tar.gz -C ${{ matrix.language }}/ .
+          - name: Upload NodeJS artifacts
+            if: matrix.language == 'nodejs'
+            uses: actions/upload-artifact@v2
+            with:
+                name: ${{ matrix.language  }}-sdk.tar.gz
+                path: ${{ github.workspace}}/${{ matrix.language }}.tar.gz
+          - name: Compress Other Lang SDK folder
+            if: matrix.language != 'nodejs'
+            run: tar -zcf sdk/${{ matrix.language }}.tar.gz -C sdk/${{ matrix.language }}/ .
+          - name: Upload Other Lang artifacts
+            if: matrix.language != 'nodejs'
+            uses: actions/upload-artifact@v2
+            with:
+                name: ${{ matrix.language  }}-sdk.tar.gz
+                path: ${{ github.workspace}}/sdk/${{ matrix.language }}.tar.gz
+        strategy:
+            fail-fast: true
+            matrix:
+              language:
+                - nodejs
+                # - python
+                # - dotnet
+                # - go
+                # - java
```

Comment:
doesn't make a lot of sense to run a matrix with only one value

Reply:
Not yet, but hopefully we'll support upgrade tests in other languages at some point (pulumi/providertest#23). Happy to remove the commented languages but I'd leave the matrix structure in place.

Reply:
ok, that seems reasonable. Maybe just add a comment that points to the GH issue for expanding testing to other languages?